apps/src/p5lab/redux/locationPicker.js

```diff
@@ -14,20 +14,26 @@ export default function locationPicker(state, action) {
   switch (action.type) {
     case REQUEST_LOCATION:
       return {
+        ...state,
         mode: LocationPickerMode.SELECTING,
-        lastSelection: undefined
+        lastSelection: undefined,
+        requestTime: Date.now()
       };
     case CANCEL_LOCATION_SELECTION:
       return {
-        mode: LocationPickerMode.IDLE
+        ...state,
+        mode: LocationPickerMode.IDLE,
+        lastSelection: undefined
```

Comment:
I don't have too much context, but assume all the changes to this file became necessary?

Reply:
Yeah, because I want the requestTime to persist in the state even if other events happen, so I'm using the spread operator and overwriting properties rather than creating a fully new state object